### ISSUE TITLE
Keypoint RCNN predicted probabilities greater than 1?

### ISSUE THREAD

**Issue by alessiosarullo** [facebookresearch/detectron2#319]
I am performing inference using a pretrained model (keypoint_rcnn_R_101_FPN_3x). Given an input image, the keypoint output should be N x K x 3, where N is the number of detected persons, K=17 is the number of keypoints and each keypoint should be (x, y, prob) according to the documentation of function `keypoint_rcnn_inference` in `detectron2/modeling/roi_heads/keypoint_head.py`. However, some of the "probabilities" I get are above 1. Did I misunderstand something? Is this expected?

**Comment by ppwwyyxx** [comment]
We may be returning the logits (before sigmoid()) rather than the probabilities.
Thanks for noticing this. We'll consider changing it.

**Comment by ppwwyyxx** [comment]
After checking the code, it is returning neither logits or probabilities. The outputs are just scores that are normalized to probabilities, but normalized to become more comparable:
https://github.com/facebookresearch/detectron2/blob/08f2c98286c7dd033f9446ac3bf8f7280eee1afd/detectron2/structures/keypoints.py#L181-L185

So we'll address the issue by updating the documentation.

**Comment by cm107** [comment]
@ppwwyyxx 
Would it be correct to say that the scores have no maximum value?
I can somewhat figure out what a good keypoint score threshold is through trial and error, but it's hard for me to determine, overall, whether or not the keypoint model is sufficiently trained because the scores do not seem to approach any maximum.

**Comment by ppwwyyxx** [comment]
It has a maximum. The maximum depends on the details of bicubic implementation.

**Comment by cm107** [comment]
@ppwwyyxx Is there a way to calculate this maximum?
If possible I would like rescale the score on a range of 0 to 1.
Is there a good reason why it isn't rescaled between 0 to 1 by default?

**Comment by ppwwyyxx** [comment]
In theory there is, but I cannot help with it.
I don't think knowing a maximum is useful for anything. You can assume the maximum is 10 because I think the maximum is smaller than that.

**Comment by udithhaputhanthri** [comment]
Hi @ppwwyyxx ,
I just wanted to clarify the below point. 

How the bi-cubic interpolation makes the score >1. Because, if we want to use pure softmax, we should use tmp_full_resolution instead of tmp_pool_resolution for the normalizing (right ?).  But considering the fact that the bi-cubic interpolation does not change the value of the maximum score of a map, Does not the maximum score always <1 even though we normalize it using an interpolated map (POOL_W x POOL_H)?

**Let me ask this in a simpler way**:
Shouldn't the maximum score over the **ROI map** is the same as the maximum score over the **corresponding interpolated map**. If so, how it should be possible for the score to be >1?

Is there anything I have missed on this

Thank you, Highly appreciated your attention

**Comment by udithhaputhanthri** [comment]
Hi @ppwwyyxx ,
I just wanted to clarify the below point. 

How the bi-cubic interpolation makes the score >1. Because, if we want to use pure softmax, we should use tmp_full_resolution instead of tmp_pool_resolution for the normalizing (right ?).  But considering the fact that the bi-cubic interpolation does not change the value of the maximum score of a map, Does not the maximum score always <1 even though we normalize it using an interpolated map (POOL_W x POOL_H)?

**Let me ask this in a simpler way**:
Shouldn't the maximum score over the **map (W x H)** is the same as the maximum score over the **corresponding interpolated map (POOL_W x POOL_H)**. If so, how it should be possible for the score to be >1 even we use the interpolated map for the normalizing?

Is there anything I have missed on this

Thank you, Highly appreciated your attention

**Comment by fortminors** [comment]
Is there a way to provide some value in [0,1] during training phase and predict it as a keypoints score?
